### PR TITLE
Add KEDA cooldownPeriod/pollingInterval property for Container App & Jobs

### PR DESCRIPTION
**Pull request by tawalke**
# ARM (Control Plane) API Specification Update Pull Request 

## Files Modified for PR
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/Job_CreateorUpdate_EventTrigger.json

_**Updated: modified files - examples**_
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_CreateOrUpdate_ConnectedEnvironment.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_CreateOrUpdate.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_Get.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_Kind_CreateOrUpdate.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_ListByResourceGroup.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_ListBySubscription.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_ManagedBy_CreateOrUpdate.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_Patch.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_SourceToCloudApp_CreateOrUpdate.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_Start.json
- specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/ContainerApps_TcpApp_CreateOrUpdate.json

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [X] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by jijohn14 — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 1]
```diff
@@ -967,6 +967,11 @@
       "format": "int32",
       "type": "integer"
     },
+    "CooldownPeriod": {
```
[the remark this answers is not in the document]
Comment (tawalke) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:970]: Updated CommonDefinitions file to add both properties of cooldownPeriod and pollingInterval in commit https://github.com/Azure/azure-rest-api-specs/pull/30154/commits/c23a4cf01a2da0fc7ddebee75cdc65a8886ac788
image

**Review by JennyLawrance — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json [hunk 2]
```diff
@@ -590,6 +590,18 @@
           "type": "integer",
           "default": 10
         },
+        "cooldownPeriod": {
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json:593]: The examples need to be updated for the new properties.

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 3]
```diff
@@ -967,6 +967,11 @@
...
+      "description": "Scaling cooldown period in seconds. Defaults to 300",
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:971]:
```suggestion
      "description": "Scaling cooldown period in seconds. Defaults to 300 seconds",
```

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json [hunk 4]
```diff
@@ -590,6 +590,18 @@
...
+          "format": "int32",
+          "description": "Optional. KEDA Cooldown Period. Defaults to 300 if not set.",
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json:595]:
```suggestion
          "description": "Optional. KEDA Cooldown Period. Defaults to 300 seconds if not set.",
```

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json [hunk 5]
```diff
@@ -590,6 +590,18 @@
...
+          "type": "integer",
+          "default": 300
+        },
+        "pollingInterval": {
+          "format": "int32",
+          "description": "Optional. KEDA Polling Interval. Defaults to 30 if not set.",
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json:601]:
```suggestion
          "description": "Optional. KEDA Polling Interval. Defaults to 30 seconds if not set.",
```

**Review by jijohn14 — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json (hunk 4 above)
Comment (tawalke) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json]: Updated description per request in commit https://github.com/Azure/azure-rest-api-specs/pull/30154/commits/d3a73a221d18a474a52515f11434662db96627fa

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json (hunk 5 above)
Comment (tawalke) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json]: Updated description per request in commit: https://github.com/Azure/azure-rest-api-specs/pull/30154/commits/d3a73a221d18a474a52515f11434662db96627fa

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json (hunk 2 above)
Comment (tawalke) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json:593]: Updated examples for every Container App example with scale. In commit: https://github.com/Azure/azure-rest-api-specs/pull/30154/commits/d3a73a221d18a474a52515f11434662db96627fa

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json (hunk 3 above)
Comment (jijohn14) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:971]: The changes are missing for this comment

**Review by jijohn14 — APPROVED** [review]

**Review by anandanthony — APPROVED** [review]

**Comment by msyyc** [comment]
sdk-suppressions of Python will be added in https://github.com/Azure/azure-rest-api-specs/pull/30156

**Pull request merged**